GrassChoppers/Modules/Home/HomeLogicController.swift

```diff
@@ -25,11 +26,21 @@ class HomeLogicController: HomeLogicControlling {
         self.dataManager.delegate = self
     }
     
+    func viewDidLoad() {
+        
+        dataManager.requestInitialState()
+        
+    }
+    
     func buttonPressed() {
         router.routeToNextScreen()
     }
 }
 
 extension HomeLogicController: HomeDataManagingDelegate {
-    
+    func didFetch(name: String) {
+        print("logic controller got the response: \(name)")
+        
+        delegate?.update(name: name)
+    }
```

Comment:
Here, `HomeLogicController` is conforming to the `HomeDataManagingDelegate` protocol so that it can act as the data manager's delegate. This enables the data manager to send responses to the logic controller.

Swift forces us to adopt the functions defined in `HomeDataManagingDelegate`, so we now have the `didFetch(name:)` function that the data manager can call to deliver the response.

In this case, we (logic controller) is the data manager's delegate, we're taking in the `name`, and forward it to _our_ delegate (view controller) via the `update(name:)` function.